internal/receiver/discoveryreceiver/statement_evaluator.go

```diff
@@ -0,0 +1,220 @@
+				return nil
+			}
+		}
+	}
+
+	if pLogs := se.evaluateStatement(statement); pLogs.LogRecordCount() > 0 {
+		se.pLogs <- pLogs
+	}
+
+	return nil
+}
+
+// Sync is a zapcore.Core method.
+func (se *statementEvaluator) Sync() error {
+	return nil
+}
+
+// evaluateStatement will convert the provided statussources.Statement into a plog.LogRecord
+// and match it against the applicable configured ReceiverEntry's status Statement.[]Match
+func (se *statementEvaluator) evaluateStatement(statement *statussources.Statement) plog.Logs {
+	se.logger.Debug("evaluating statement", zap.Any("statement", statement))
+	statementLogRecord := statement.ToLogRecord()
+
+	pLogs := plog.NewLogs()
+	receiverID, endpointID := statussources.ReceiverNameToIDs(statementLogRecord)
+	if receiverID == statussources.NoType || endpointID == "" {
+		// statement evaluation requires both a populated receiver.ID and EndpointID
+		se.logger.Debug("unable to evaluate statement from receiver", zap.String("receiver", receiverID.String()))
+		return pLogs
+	}
+
+	rEntry, ok := se.config.Receivers[receiverID]
+	if !ok {
+		se.logger.Info("No matching configured receiver for statement status evaluation", zap.String("receiver", receiverID.String()))
+		return pLogs
+	}
+
+	if rEntry.Status == nil {
+		return pLogs
+	}
+
+	stagePLogs := plog.NewLogs()
+	rLog := stagePLogs.ResourceLogs().AppendEmpty()
+	rAttrs := rLog.Resource().Attributes()
+	fromAttrs := pcommon.NewMap()
+	fromAttrs.PutString(statussources.ReceiverTypeAttr, string(receiverID.Type()))
+	fromAttrs.PutString(statussources.ReceiverNameAttr, receiverID.Name())
+	fromAttrs.PutString(statussources.EndpointIDAttr, string(endpointID))
+	se.correlateResourceAttributes(fromAttrs, rAttrs, se.correlations.GetOrCreate(receiverID, endpointID))
+	rAttrs.PutString(eventTypeAttr, statementMatch)
+	rAttrs.PutString(receiverRuleAttr, rEntry.Rule)
+	logRecords := rLog.ScopeLogs().AppendEmpty().LogRecords()
+
+	var matchFound bool
+
+	body := statementLogRecord.Body().AsString()
+	for status, matches := range rEntry.Status.Statements {
+		for _, match := range matches {
+			if shouldLog, err := se.evaluateMatch(match, body, status, receiverID, endpointID); err != nil {
+				se.logger.Info(fmt.Sprintf("Error evaluating %s statement match", status), zap.Error(err))
+				continue
+			} else if !shouldLog {
+				continue
+			}
+			matchFound = true
+			logRecord := logRecords.AppendEmpty()
+			var desiredRecord LogRecord
+			if match.Record != nil {
+				desiredRecord = *match.Record
+			}
+			statementLogRecord.CopyTo(logRecord)
+			if desiredRecord.Body != "" {
+				logRecord.Body().SetStringVal(desiredRecord.Body)
+			}
+			if len(desiredRecord.Attributes) > 0 {
+				for k, v := range desiredRecord.Attributes {
+					logRecord.Attributes().PutString(k, v)
+				}
+			}
+			severityText := desiredRecord.SeverityText
+			if severityText == "" {
+				severityText = logRecord.SeverityText()
+				if severityText == "" {
+					severityText = "info"
```

Comment:
Spec suggests it is uppercase: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#displaying-severity
```suggestion
					severityText = "INFO"
```

Reply:
updated, thanks